### PR TITLE
updated friend model to be ManyToOne, and changed the tables. Updated…

### PR DESCRIPTION
**Pull request by multiplesnakes**
… "usersearch" to be "users" in navbar, profile, and app. Added friendslist that is not yet functional

### CODE REVIEW

File: demo/src/main/java/com/example/data/FriendRepository.java [hunk 1]
```diff
@@ -6,6 +6,7 @@
 import org.springframework.stereotype.Repository;
 
 import java.util.List;
+import java.util.Optional;
```
Comment (gbreternitz) [inline demo/src/main/java/com/example/data/FriendRepository.java:9]: Is there a reason to add this? Doesn't look like it get's used.

File: demo/src/main/java/com/example/models/Friend.java [hunk 2]
```diff
@@ -12,14 +12,11 @@ public class Friend {
...
-    @Column(name = "created_date")
-    private Date createdDate;
-
-    @OneToOne(cascade = CascadeType.ALL)
+    @ManyToOne(cascade = CascadeType.ALL)
     @JoinColumn(name = "first_user_id", referencedColumnName = "id")
     private User firstUser;
 
-    @OneToOne(cascade = CascadeType.ALL)
+    @ManyToOne(cascade = CascadeType.ALL)
```
Comment (gbreternitz) [inline demo/src/main/java/com/example/models/Friend.java:19]: I'm not understanding the firstUser and secondUser fields here - is the idea to only allow two friends?

Friend.java should be a ManyToOne relationship with User, like this:

```
@ManyToOne(cascade = CascadeType.ALL)
private User user;
```

and User should have a field like:

```
    @OneToMany(mappedBy = "user", cascade = CascadeType.ALL)
    private List<Friend> friends= new ArrayList<>();
```
    
That way User can have as many friends as they like, right?

**Review by gbreternitz — CHANGES_REQUESTED** [review]
I think the Friends class needs to be reworked with a relationship similar to how Users and Watchlists are related -- seems like the current design only allows two friends, and I'm confused about how it relates to the User class twice with the firstUser and secondUser fields

File: demo/src/main/java/com/example/data/FriendRepository.java (hunk 1 above)
Comment (multiplesnakes) [reply demo/src/main/java/com/example/data/FriendRepository.java:9]: it was an artifact, can remove

File: demo/src/main/java/com/example/models/Friend.java (hunk 2 above)
Comment (multiplesnakes) [reply demo/src/main/java/com/example/models/Friend.java:19]: this has been updated!

**Comment by gbreternitz** [comment]
closed as marisa is reworking her backend code

**Pull request closed without merging**